Fix broken contribution stats upload job

PaliC opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #96003

Actually allows uploads to S3 to work properly for external contribution stats. 
Test: https://github.com/pytorch/pytorch/actions/runs/4347343883/jobs/7594534296

PaliC committed:
Fix broken contribution stats upload job

[ghstack-poisoned]

PaliC committed:
Update upload-contrib-stats.yml

PaliC committed:
Update on "Fix broken contribution stats upload job"




[ghstack-poisoned]

PaliC committed:
Update upload-contrib-stats.yml

kit1980 commented on .github/workflows/upload-contrib-stats.yml:
```diff
@@ -1,14 +1,16 @@
 name: Upload contribution stats
 
 on:
+  pull_request:
```
Why run on pull_request?

PaliC replied:
Just for testing

kit1980 commented on .github/workflows/upload-contrib-stats.yml:
```diff
@@ -1,14 +1,16 @@
+
```
Remove empty line?

huydhn commented on .github/workflows/upload-contrib-stats.yml:
```diff
@@ -22,6 +24,7 @@ jobs:
         run: |
-          echo "Uploading external contribution stats for $(date -v-1d +%F)"
-          python3 -m tools.stats.upload_external_contrib_stats --startDate "$(date -v-1d +%F)"
+          echo "Uploading external contribution stats for" "$(date -d yesterday '+%Y-%m-%d')"
```
Per our discussion, the self-hosted runner has access to some S3 buckets because of its IAM role.  You would need to add the new bucket into the list to grant the runner access.  There is no AWS key setup here in the workflow

PaliC committed:
Update upload-contrib-stats.yml

PaliC committed:
Update upload-contrib-stats.yml

PaliC committed:
Update upload-contrib-stats.yml

PaliC committed:
Update upload-contrib-stats.yml

remove pull request trigger

huydhn commented on .github/workflows/upload-contrib-stats.yml:
```diff
@@ -1,14 +1,19 @@
     - cron:  37 7 * * *
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.sha }}-${{ github.event_name == 'workflow_dispatch' }}
```
I think you mean `-${{ github.event_name == 'schedule' }}` here as this workflow only run on schedule (after the `pull_request` part is removed per the above comment

PaliC replied:
Gotcha, I just copied what was in the error message haha

huydhn reviewed (approved):
LGTM! Only a small comment on the concurrency group and to remove `pull_request` before committing

PaliC committed:
Update upload-contrib-stats.yml

PaliC committed:
Update upload-contrib-stats.yml

PaliC commented:
@pytorchbot merge -g

PaliC committed:
Update on "Fix broken contribution stats upload job"


Actually allows uploads to S3 to work properly for external contribution stats. 
Test: https://github.com/pytorch/pytorch/actions/runs/4347343883/jobs/7594534296


[ghstack-poisoned]

PaliC commented:
@pytorchbot merge -g
